src/AzureExtension/DevBox/DevBoxInstance.cs

```diff
@@ -541,6 +562,165 @@ public IAsyncOperation<ComputeSystemOperationResult> ModifyPropertiesAsync(strin
+            Process? p = Process.Start(psi);
+            if (p != null)
+            {
+                p.WaitForExit();
+                var exitcode = p.ExitCode;
+                if (exitcode == ExitCodeSuccess)
+                {
+                    return new ComputeSystemOperationResult();
+                }
+            }
+
+            return new ComputeSystemOperationResult(new NotSupportedException(), Resources.GetResource(Constants.DevBoxUnableToPerformOperationKey), "The Operation Failed");
+        }).AsAsyncOperation();
+    }
+
+    public IAsyncOperation<ComputeSystemPinnedResult> GetIsPinnedToStartMenuAsync()
+    {
+        return Task.Run(() =>
+        {
+            var psi = new ProcessStartInfo();
+            psi.UseShellExecute = true;
+            psi.FileName = string.Format(CultureInfo.InvariantCulture, ProtocolPinString, "startmenu", "status", WorkspaceId, DisplayName, Username, Environment);
+            Process? p = Process.Start(psi);
+            if (p != null)
+            {
+                p.WaitForExit();
+                var exitcode = p.ExitCode;
+                if (exitcode == ExitCodePinned)
+                {
+                    return new ComputeSystemPinnedResult(true);
+                }
+                else if (exitcode == ExitCodeUnpinned)
+                {
+                    return new ComputeSystemPinnedResult(false);
+                }
+            }
+
+            return new ComputeSystemPinnedResult(new NotSupportedException(), Resources.GetResource(Constants.DevBoxUnableToPerformOperationKey), "The Operation Failed");
+        }).AsAsyncOperation();
+    }
+
+    public IAsyncOperation<ComputeSystemPinnedResult> GetIsPinnedToTaskbarAsync()
+    {
+        return Task.Run(() =>
+        {
+            var psi = new ProcessStartInfo();
+            psi.UseShellExecute = true;
+            psi.FileName = string.Format(CultureInfo.InvariantCulture, ProtocolPinString, "taskbar", "status", WorkspaceId, DisplayName, Username, Environment);
+            Process? p = Process.Start(psi);
+            if (p != null)
+            {
+                p.WaitForExit();
+                var exitcode = p.ExitCode;
+                if (exitcode == ExitCodePinned)
+                {
+                    return new ComputeSystemPinnedResult(true);
+                }
+                else if (exitcode == ExitCodeUnpinned)
+                {
+                    return new ComputeSystemPinnedResult(false);
+                }
+            }
+
+            return new ComputeSystemPinnedResult(new NotSupportedException(), Resources.GetResource(Constants.DevBoxUnableToPerformOperationKey), "The Operation Failed");
+        }).AsAsyncOperation();
+    }
+
+    public async void LoadWindowsAppParameters()
+    {
+        if (RemoteConnectionData is null)
+        {
+            await GetRemoteLaunchURIsAsync(new Uri(DevBoxState.Uri));
+        }
+
+        var uriString = RemoteConnectionData?.RdpConnectionUrl;

```

Comment:
CloudPcConnectionUrl is available in under RemoteConnectionData already. Let's use that.